controller_manager:
  ros__parameters:
    update_rate: 1000
    use_sim_time: true

    go2_joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    go2_joint_trajectory_controllers:
      type: joint_trajectory_controller/JointTrajectoryController


go2_joint_state_broadcaster:
  ros__parameters:
    publish_rate: 100


go2_joint_trajectory_controllers:
  ros__parameters:
    joints:
      # - FR_hip_joint
      # - FL_hip_joint
      # - RR_hip_joint
      # - RL_hip_joint
      # - FR_thigh_joint
      # - FL_thigh_joint
      # - RR_thigh_joint
      # - RL_thigh_joint
      # - FR_calf_joint
      # - FL_calf_joint
      # - RR_calf_joint
      # - RL_calf_joint
      - lf_hip_joint
      - lf_upper_leg_joint
      - lf_lower_leg_joint
      - rf_hip_joint
      - rf_upper_leg_joint
      - rf_lower_leg_joint
      - lh_hip_joint
      - lh_upper_leg_joint
      - lh_lower_leg_joint
      - rh_hip_joint
      - rh_upper_leg_joint
      - rh_lower_leg_joint 

    command_interfaces:
      - effort
      # instead of effort, for ease of use.
    state_interfaces:
      - position
      - velocity

    gains:
      FR_hip_joint  : {p: 5000.0, d: 10.0, i: 1.0, i_clamp: 10.0, ff_velocity_scale: 1.0}
      FL_hip_joint  : {p: 5000.0, d: 10.0, i: 1.0, i_clamp: 10.0, ff_velocity_scale: 1.0}
      RR_hip_joint  : {p: 5000.0, d: 10.0, i: 1.0, i_clamp: 10.0, ff_velocity_scale: 1.0}
      RL_hip_joint  : {p: 5000.0, d: 10.0, i: 1.0, i_clamp: 10.0, ff_velocity_scale: 1.0}
      FR_thigh_joint: {p: 5000.0, d: 10.0, i: 0.1, i_clamp: 1.0, ff_velocity_scale: 1.0}
      FL_thigh_joint: {p: 5000.0, d: 10.0, i: 0.1, i_clamp: 1.0, ff_velocity_scale: 1.0}
      RR_thigh_joint: {p: 5000.0, d: 10.0, i: 0.1, i_clamp: 1.0, ff_velocity_scale: 1.0}
      RL_thigh_joint: {p: 5000.0, d: 10.0, i: 0.1, i_clamp: 1.0, ff_velocity_scale: 1.0}
      FR_calf_joint : {p: 5000.0, d: 10.0, i: 0.1,  i_clamp: 1.0, ff_velocity_scale: 1.0}
      FL_calf_joint : {p: 5000.0, d: 10.0, i: 0.1,  i_clamp: 1.0, ff_velocity_scale: 1.0}
      RR_calf_joint : {p: 5000.0, d: 10.0, i: 0.1,  i_clamp: 1.0, ff_velocity_scale: 1.0}
      RL_calf_joint : {p: 5000.0, d: 10.0, i: 0.1,  i_clamp: 1.0, ff_velocity_scale: 1.0}

    stt.0e_publish_rate: 100.0 # Defaults to 50
    action_monitor_rate: 20.0 # Defaults to 20
